invenio_damap/services/services.py

```diff
@@ -33,68 +36,70 @@ def schema(self):
-
-        return headers
-
-    def _get_user_id(self, identity):
-        """Get the user id from the identity."""
-        return self.config.damap_person_id_function(identity=identity)
-
-    def add_record_to_dmp(self, identity, recid, dmp_id, data):
-        """Add the provided record to the DMP"""
-
-        person_id = self._get_user_id(identity=identity)
-        headers = self._create_headers(identity)
-
-        # this will also perform permission checks, ensuring the user may access the record.
-        record = current_rdm_records_service.read(identity, recid)
-        exported_record = InvenioDAMAPExport.export_as_madmp(record, **data)
-
-        r = requests.post(
-            url=self.config.damap_base_url
-            + "/api/invenio-damap/dmps/{}/{}".format(dmp_id, person_id),
-            headers=headers,
-            json=exported_record,
+    def _create_auth_jwt(self, identity, expires_in=600):
+        """Creates an authorization jwt token for DAMAP."""
+        person_data = self.config.damap_person_function(identity)
+
+        if person_data is None:
+            return
+
+        # more info about JWT claims: https://www.rfc-editor.org/rfc/rfc7519.html#section-4
+        return jwt.encode(
+            {
+                "sub": str(current_user.id),
+                "exp": int(time()) + expires_in,
+                "iat": int(time()),
+                "invenio-damap": {
+                    "identifiers": {
+                        **person_data,
+                    }
+                },
+            },
+            self.config.damap_shared_secret,
+            algorithm="HS256",
         )
 
-        r.raise_for_status()
-
-        return record
+    def _create_headers(self, identity, *args, **kwargs):
+        """Creates the auth header and additional ones, if defined."""
+        headers = self.config.damap_custom_header_function(identity=identity)
+        return headers
 
-    def search(self, identity, params):
+    def search(self, identity, params, jwt=None, **kwargs):
         """Perform search for DMPs."""
         self.require_permission(identity, "read")
 
         search_params = self._get_search_params(params)
-        person_id = self._get_user_id(identity=identity)
         headers = self._create_headers(identity)
+        headers.update(
+            {"X-Auth": (f"{jwt}" if jwt else f"{self._create_auth_jwt(identity)}")}
+        )
 
         r = requests.get(
-            url=self.config.damap_base_url
-            + "/api/invenio-damap/dmps/person/{}".format(person_id),
+            url=self.config.damap_base_url + "/api/madmps",
             headers=headers,
             params=search_params,
         )
-
-        r.raise_for_status()
+        # We don't want to trigger the exceptions on these codes.
+        # These indicate no valid identity between the 2 systems or no DMPs.
+        if r.status_code not in [401, 403, 404]:
```

Comment:
401 should be unauthenticated (do not know who you are. Either invalid authentication schema or wrong credentials (i.e. could not resolve identity))
403 should be unauthorized (i know who you are, but you are not allowed to perform this action)
404 should be not found (either endpoint or requested resource is not found. Since the search endpoint should not return a 404, it can only be the endpoint. Therefore a 404 would make sense to also raise).

DAMAP should return a:
- 401 if the no valid authentication schema is provided, identity could not be resolved or multiple identities resolve to different DMPs.
- 403 if the identity can be resolved, but this identity is not allowed to update a DMP
- 404 if an endpoint is not found, a resource is not available (i.e. wrong DMP id when adding a dataset), or to mask a 403 if the existence of a resource should not be revealed)

Will add this to the DAMAP PR as well.